  <launch>

  <!-- Inputs -->
  <arg name="imu_data_topic"        default="/platform/imu/data" />
  <arg name="point_cloud_topic"     default="/platform/velodyne_points" />

  <!-- Outputs -->
  <arg name="slam_odometry_topic"   default="/slam/s_graphs/odom_to_map" />
  <arg name="slam_markers_topic"    default="/slam/s_graphs/markers" />
  <arg name="slam_map_points_topic" default="/slam/s_graphs/map_points" />
  
  <arg name="msf_enabled"          default="false" />
  <arg name="april_tag_enabled"    default="true" />
  <arg name="base_link_frame"      default="base_footprint" />
 
  <!-- HDL GRAPH SLAM -->
  <node pkg="nodelet" type="nodelet" name="s_graphs_nodelet_manager" args="manager" output="screen" />
  <node pkg="nodelet" type="nodelet" name="s_graphs" args="load s_graphs/HdlGraphSlamNodelet s_graphs_nodelet_manager" output="screen">
    <rosparam command="load" file="$(find s_graphs)/config/s_graphs.yaml" />
    <param if="$(arg april_tag_enabled)"     name="wait_trans_odom2map"  value="true" />
    <param unless="$(arg april_tag_enabled)" name="wait_trans_odom2map"  value="false" />
    <param name="odom_frame_id"              value="odom_reset" />
    <param name="map_frame_id"               value="map" />
    <param name="odom_frame_id"              value="odom_reset" />
    <!-- Inputs -->
    <remap from="/odom"                       to="/platform/reset_odometry" />
    <remap from="/filtered_points"            to="/slam/hdl_prefilter/filtered_point_cloud" />
    <remap from="/odom2map/initial_pose"      to="/odom_to_map/initial_pose" />
    <!-- Outputs -->
    <remap from="/s_graphs/odom2map"    to="$(arg slam_odometry_topic)" />
    <remap from="/s_graphs/markers"     to="$(arg slam_markers_topic)" />
    <remap from="/s_graphs/map_points"  to="$(arg slam_map_points_topic)" />
    <remap from="/s_graphs/read_until"  to="s_graphs/read_until" />
  </node>
  
  <node pkg="s_graphs" name="map2odom_publisher" type="map2odom_publisher.py">
    <param name="odom_frame_id" value="odom_reset" />
    <param name="map_frame_id" value="map" />
    <remap from="/s_graphs/odom2map"  to="/slam/s_graphs/odom_to_map" />
  </node>

</launch>  